### ISSUE TITLE
FIX: Failed to bridge OMG L1->L2 for some users/wallets

### ISSUE THREAD

**Issue by Ralph89** [omgnetwork/optimism#501]
So, I am trying to brige my OMG token.
The first tx was for approving the spend, that was succesfull. Afterwards I got to metamask popup to send the tokens to the boba l2.
I did decline that tx (high gas fees)
Now a bit later I wanted to bridge the tokens but now I just keep getting the popup Failed to brige erc20.
When I open up the browser debug it shows me this:
![boba](https://user-images.githubusercontent.com/3372661/134140166-03abb326-f9b0-4f37-9bb1-2accebc6a42e.png)

**Comment by lowballer356** [comment]
same issue here, I was able to bridge eth to l2 on the wallet with the error for omg. I was also able to bridge both eth and omg from another wallet,   Gas is nuts.......  over$140 per wallet, ate up a fair portion of my small stack.

**Comment by Ralph89** [comment]
Yea, I was thinking about sending my omg to another wallet to do it from there if it is not resolved. still something to look into for them :)

**Comment by lowballer356** [comment]
my plan was to send and stake out of a couple wallets and see what happens with the air drop they announced, so I would like to know how to get past the failure.  I also canceled when i saw the gas fees for OMG.  It was around 120 and I only had 60 total OMG  Almost half of the value lost to gas.  Guess that is why we are all hunting layer 2.

**Comment by CAPtheorem** [comment]
Investigating now. The best option for cost effective on ramping is probably the Fast Bridge. There will be fixes / action on this with 24 hours. Yup gas is insane.

**Comment by Hellprdel** [comment]
I have the same problem. When I accepted fee for bridge so metamask notification sent me that transaction is not accepted but etherscan is ok, so I don´t know what next. When I try same again so bobanetwork write me "Failed to brige erc20"

**Comment by jwboba** [comment]
> Investigating now. The best option for cost effective on ramping is probably the Fast Bridge. **There will be fixes / action on this within 24 hours.** Yup gas is insane.
> 
> Anecdotal observation: some people _are_ able to bridge OMG L1->L2 - this issue only seems to affect a subset of users/wallets.

From what I've seen, if you cancel the transaction (cause users saw the high gas costs) or the transaction fails the first time, the wallet won't work anymore.

**Comment by Hellprdel** [comment]
I don't really understand why it creates something somewhere and it doesn't work 100%. Only the transfers cost me 30 USD and in the end it doesn't work again. Is there any competent person to whom I will send the link from etherscan and will be able to explain and put this into operation?

**Comment by CAPtheorem** [comment]
We have a partial fix. The problem is that some users were using an outdated firmware for their Ledger hardware wallets. 

https://github.com/omgnetwork/optimism/pull/496/commits/1c792e3c2c5cb4b34c3c4ad5f120d68393209562

**Comment by Hellprdel** [comment]
I'm using the metamask I downloaded today, so it's hard to be an outdated version

**Comment by CAPtheorem** [comment]
The above fix is specific for people using the **Ledger hardware wallet** with hardware firmware < 2.0.0

**Comment by Hellprdel** [comment]
I wonder how much I get in the airdrop. Normally it is enough to send the address of the wallet to the date and nothing has to be transferred anywhere. I pay the transfer fees here and it still doesn't work. What's the point of inventing something new when it doesn't work and it's not tested?

**Comment by Hellprdel** [comment]
It doesn't work for $ 20, so here you advise me to transfer it for several times higher fee and hope that it will be transferred and if not, someone will deal with it preferentially? What is this nonsensical advice?

**Comment by Hellprdel** [comment]
So what should I do now?
1. send it fast for nonsense, which I certainly won't do when I see it here ...
2. Wait for someone to fix it?
When will someone fix it?

**Comment by CAPtheorem** [comment]
As noted above, we are working on this right now.

1/ Bridge issues for users of Ledger hardware wallet below 2.0.0 = *resolved* - please update firmware
2/ Bridge issues for some users of MetaMask version below  10.1.0 = *resolved* - please update MetaMask
3/ Bridge issues for some users of MetaMask version ===  10.1.0 = *open* - investigating

**Comment by Ralph89** [comment]
@CAPtheorem I just arrived home from work, I just tried it with a different pc. I imported the account where I have my omg tokens on into metamask 10.1.0. 

I do still get the error Failed to bridge erc20 token.
Also to clarify the steps I did:
1: Approve the spend tx
2: decline the tx that popped up to send my omg tokens to the boba layer  (high tx fees) 
I am able to transfer eth to the boba layer it only happens with the omg tokens (did not try any other erc20 token tho)

Please let me know if I can help you further

![image](https://user-images.githubusercontent.com/3372661/134227053-0850a00a-bfb0-41a4-b66b-609cb3196676.png)

**Comment by Hellprdel** [comment]
> As noted above, we are working on this right now.
> 
> 1/ Bridge issues for users of Ledger hardware wallet below 2.0.0 = _resolved_ - please update firmware
> 2/ Bridge issues for some users of MetaMask version below 10.1.0 = _resolved_ - please update MetaMask
> 3/ Bridge issues for some users of MetaMask version === 10.1.0 = _open_ - investigating

I have version 10.1.0 I will wait

**Comment by Hellprdel** [comment]
How to unconnected metamask wallet from bobanetwork?

**Comment by Hellprdel** [comment]
I tried uninstall metamask from chrome a install again but nothing - still the same "Failed to brige erc20" I have 3 transactions 2 ok with etherscan a one fail. It is interesting that I sent the conversion only once, because it did not work several times, but in etherscan I have 3 outgoing transactions, where I was always deducted a fee. It really doesn't work very well.

**Comment by nocturnien** [comment]
I have read the above comments and have the same issue, able to approve but not bridge.

I have Ledger 2.0.0 and MM 10.1.0  
 
I have used my Ledger successfully yesterday, so obviously less than 1 yr since last use.  
 
I have enabled contract data.  

My approval happened successfully, although I did set my gas price to 76 gwei, so it took a bit of wait before it cleared mempool.  And while Etherscan showed that it cleared, my MM kept showing that the txn was pending.  I had to close the browser and launch again in order for MM showing the approve under activity.

When I tried to bridge, it failed immediately.  I tried several times

**Comment by CAPtheorem** [comment]
*The root cause has been found, and a fix in in the works.* Briefly, it relates to the OMG token's nonstandard configuration relative to most other tokens, based on contract architecture choices made in June of 2017.

**Comment by nocturnien** [comment]
Thx.  So if I have already successfully approved once for a bridge amt, do I have to go back and do all 3 steps or just steps 1 and 3?

**Comment by CAPtheorem** [comment]
It depends. For example, if you approved 10, but want to bridge 7, you are all set. But if you approved 3, and want to bridge 7, then three steps will be needed. 3->0, 0->7, bridge. This is out of our control, and a basic feature of the OMG token contract.

**Comment by nocturnien** [comment]
What do you mean by all set?  Just now I tried to bridge an amount less than what was successfully approved earlier today.  It gave me the same erc 20 error.  

Do I need to approve 0, and then approve all, and then bridge?  

The above instructions and your responses are a bit confusing. 

Thx

**Comment by CAPtheorem** [comment]
we are about 3 hours away from actual production

**Comment by nocturnien** [comment]
Oh, got it.  Sorry for jumping the gun

**Comment by nocturnien** [comment]
Saw the screen shot and the sentence "fix moving to production" and thought it is a done deal

**Comment by CAPtheorem** [comment]
sorry - that means we have everything working locally, and are in the process of moving it to production

**Comment by lowballer356** [comment]
My metamask wallet is pretty new.  This year.  I also bridged some eth off the same wallet and that went through with no issues.  I have a secrod metamask wallet that both eth and omg went through with no problems.  Same metamask acct so it can't be an issue with the firmware.  It is a bummer that I burned a shit ton of expensive eth trying to get bridged with the approvals. I know you folks are testing and bugs come out.    Thanks for the help.  I'm going to try to bridge omg again tomorrow night.  Between the eth and omg and approvals I'm at about 140 USD per wallet. So between the two I'm looking at nearly 300usd That is to bridge 60 omg and ..07 eth between the two.    Makes it tough for the small fish to be part of defi

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by RogerThat002** [comment]
> 
> 
> It depends. For example, if you approved 10, but want to bridge 7, you are all set. But if you approved 3, and want to bridge 7, then three steps will be needed. 3->0, 0->7, bridge. This is out of our control, and a basic feature of the OMG token contract.

Hi: A request: Once the fix IS in production, more clear, specific, and exact instructions on how we can proceed to accomplish the Bridge would be very much appreciated. I, like several others, have "Approved Spends" [2 in my case at a cost of ~$25]... but--so far--it is not at all clear IF we can use those approvals to complete our L2 Bridge and exactly HOW [step by step process...] we can use to actually, successfully DO it ...OR--on the other hand--IF we need to initiate a whole new Bridging process from the start.  If the latter, but really in either case, as an owner of many thousand OMG, I expect I can speak for us all in stating the obvious: we'd all appreciate some compensation for our lost fees as well as our time in helping Boba Network to BetaTest the highly frustrating and largely non-functional bridging UX that you've subjected us to...Thanks for hearing this!

**Comment by CAPtheorem** [comment]
I hear you and it's been a long day for us all. Step one is to get the OMG-bridges sorted and tested. Separately, please retain your TX hashes for the unnecessary approvals. Speaking for everyone here, thank you for all the information and patience, and please keep bug reports and suggestions coming.

**Comment by CAPtheorem** [comment]
@RogerThat002 @lowballer356 @nocturnien @Hellprdel @Septboba @jwboba - please retain your TX hashes - standby for more specific instructions. We have your back.

**Comment by CAPtheorem** [comment]
Jet86 on TG or Discord is waiting for you

**Comment by jet86** [comment]
@RogerThat002 @lowballer356 @nocturnien @Hellprdel @Septboba @jwboba - please log a support ticket in our [Discord](https://omg.eco/support) by heading to the #support-boba channel and clicking "Create ticket" then entering the details, including the transaction hashes.

If you have any issues, you can reach out to me either in Discord or in Telegram.

**Comment by Septboba** [comment]
Can you guys give us a quick walk through of how to get the stuff we sent to L2 to show up. I tried finding said #support-boba channel to create a ticket and had zero luck, even sent a screen shot of what I was seeing on discord.

**Comment by Hellprdel** [comment]
I wanted to bridge yesterday. Gas charges left in etherscan's transactions are ok, but nothing happened. Subsequently, I tried and wrote to me that Failed. I tried it in the morning and the transaction went through again, but nothing moved. It already cost me about $ 70 and still nothing. Developers must be great masters if they can't do this either.

So a question for developers:

Who will refund my gas charges if it doesn't work?

Is there a well-founded person who is able to answer me, why etherscan tells me that the transaction is ok, but nothing has been transferred to L2?

**Comment by Ralph89** [comment]
Well, to be honest it seems like you are just resetting the approval, but after the txs it worked. bit of a shame as it cost money
I found a similar issue with staking the omg tokens on the l2 I will create a new ticket for it

**Comment by greyham2021** [comment]
@CAPtheorem, @jet86 I am facing the same issue. Were there more specific instructions given somewhere to walk through potential fixes? I have tried following the steps above 5+times this evening to bridge OMG, after also updating all software and firmware on ledger and metamask and still all transaction attempts are failing. Any help appreciated. Thanks

**Comment by RogerThat002** [comment]
> 
> 
> @RogerThat002 @lowballer356 @nocturnien @Hellprdel @Septboba @jwboba - please log a support ticket in our [Discord](https://omg.eco/support) by heading to the #support-boba channel and clicking "Create ticket" then entering the details, including the transaction hashes.
> 
> If you have any issues, you can reach out to me either in Discord or in Telegram.

G'morning! Thanks to the team for all your efforts to address this bug. I hear that a fix is now in production and we have a UX that delivers successful bridging...I will retest it again myself tonight and report my experience. Also, I hope the team is taking seriously the need to have a communication specialist developing detailed instructions on exactly how to Bridge--especially with noobs in mind--as I've suggested above.  We've seen how even experienced crypto investors such as myself have had a very frustrating and time-consuming experience in bridging to Boba.  I'm concerned that too many who're new to Boba and crypto will be left behind unnecessarily. That outcome would be quite unfortunate for all of us.

**Comment by RogerThat002** [comment]
> 
> 
> Can you guys give us a quick walk through of how to get the stuff we sent to L2 to show up. I tried finding said #support-boba channel to create a ticket and had zero luck, even sent a screen shot of what I was seeing on discord.

I found the #support-boba channel on Boba's Discord, which you'll want to Join, if you haven't already. https://discord.com/invite/m7NysJjKhm It's easy to create a ticket once you're there.

**Comment by Septboba** [comment]
They fixed the channel after I messaged Jet. I still have not gotten an answer back but Jet is not online so it will probably take time.

**Comment by nocturnien** [comment]
@CAPtheorem Glad to see that the fix has been pushed to production.  However, when I tried to fast bridge, I am told that the liquidity bridge only has 6.6 OMG.  When will it increase?  What would cause it to increase?  I prefer to use fastbridge since I already have spent eth on an approval (or does it not matter whether regular or fastbridge) and it is less costly?

Also, as echoed by other users here, we understand that we are early and things will break some times.  But it is very important for the team to provide solid communication and support coverage.  Your early users can help spread the word and evangelize the project if you take care of them.  On the flip side, if they get frustrated and leave, you wont get the positive network effect and possibly distractors.  Please make support/communication a strategic high priority item even though its tempting to chase after other fires that are more visible. thx

**Comment by jwboba** [comment]
I think questions should be directed onto TG and Discord for support. Github is for bug fixing. @nocturnien fast bridge liquidity is provided by the community so you just have to wait for liquidity (other users providing it).

**Comment by nocturnien** [comment]
Looks like the liquidity is back up now.  But it seems that I have to approve again even though I am trying to bridge an amount equal to what I was approved yesterday.  @CAPtheorem I thought I dont need to approve again per what you wrote yesterday as long as I am bridging an amt less than what I previously approved?

**Comment by lowballer356** [comment]
Just wanted to let you know that I was able to successfully bridge my OMG over tonight and got both my eth and OMG staked.

It did charge me the approval fee twice tonight though.

I am a total newb on etherscan so I will need to figure out how to read it and how many extra charges were incurred through unnecessary additional approvals and try to get those logged somewhere for you.  Learning opportunity for me.

I understand you are still in beta, so things happen.  I am not expecting anything to be refunded, but if that is something you can do, or if you can sweeten the airdrop (if my pittance qualifies) , I would greatly appreciate it.

With that being said, I want to personally thank you for actually looking into this, being helpful and genuinely seeming to care about helping us get our assets through the bridge.  I have worked with several other help desks for other projects and their help desk sucked balls.

If you have a supervisor, please forward this to them, you are what I am looking for as a manager and I have been very satisfied with the updates, response time, insights, and resolution.

THANK YOU!

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by lowballer356** [comment]
So, I tried to get my OMG staked last night and had the same issue with request to spend eth multiple times.  Unfortunately I wasn’t watching my balance and now my eth  balance is too low to unstake some eth, or to try to stake my OMG.  Bummer, I guess ill have to bridge some more at some point.  This is getting spendy.

Might want to look into why it is requesting that we authorize spending after spending has already been approved for a token multiple times.

Thank you.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by 0x-Stoic** [comment]
@CAPtheorem @jet86 or anyone from the team, I have the same experience as @lowballer356.  I needed to do multiple approvals, a total of 3 before I was able to bridge.  This included once before the new fix and twice after the fix.  The 2 approvals after the fix happened immediately one after another.  Very strange and costly.  Not a happy camper right now.  And certainly not happy with the lack of responses to user comments in this thread.  This is not how your interact with the early adopters in the community.  My confidence in Boba, the protocol, and the team's responsiveness is pretty low.  I am seriously considering bridging my OMG back to L1 and dump them.

**Comment by lowballer356** [comment]
Good morning.  I was asked to save my hashes for this issue.  I believe I have that list although I am a newb and might not have them correctly logged.

Where should I send that list.

Thank you

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
